java/arcs/sdk/android/storage/ServiceStore.kt

```diff
@@ -133,12 +137,15 @@ class ServiceStore<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
                     )
                 }
             }
-        })
+        }, completion)
+        return completion.get()
     }
 
     override fun off(callbackToken: Int) {
+        val completion = FutureResult()
```

Comment:
Ah, now I see why `Future` is preferred to `Deferred`... because we don't want to make `on` and `off` into suspend functions. 👍